src/app/modules/zxing-scanner/browser-code-reader.ts

```diff
@@ -153,26 +163,28 @@ export class BrowserCodeReader {
         videoElement.addEventListener('loadedmetadata', this.videoLoadedMetadataEventListener);
     }
 
-    private checkTorchCompatibility(stream: MediaStream): boolean {
-        let compatible = false;
+    private checkTorchCompatibility(stream: MediaStream): void {
+
         this.track = stream.getVideoTracks()[0];
 
         const imageCapture = new ImageCapture(this.track);
+
         const photoCapabilities = imageCapture.getPhotoCapabilities().then((capabilities) => {
-            this.torchCompatible.next(!!capabilities.torch);
-            compatible = capabilities.torch;
+            const compatible = !!capabilities.torch || ('fillLightMode' in capabilities && capabilities.fillLightMode.length !== 0);
+            this.torchCompatible.next(compatible);
```

Comment:
I think it would be nice to return the Observable, because we could make use of it in the component.

Reply:
You can get the observable with a property. It is called `torchAvailable`.

Reply:
I added the output to the component.